pkg/tagger/collectors/kubelet_extract.go

```diff
@@ -106,10 +100,33 @@ func (c *KubeletCollector) parsePods(pods []*kubelet.Pod) ([]*TagInfo, error) {
 // parseDeploymentForReplicaset gets the deployment name from a replicaset,
 // or returns an empty string if no parent deployment is found.
 func (c *KubeletCollector) parseDeploymentForReplicaset(name string) string {
-	// TODO optionaly query the cluster agent instead of assuming?
-	match := KubeReplicaSetRegexp.FindStringSubmatch(name)
-	if len(match) < 2 {
+	lastDash := strings.LastIndexAny(name, "-")
+	if lastDash == -1 {
+		// No dash
+		return ""
+	}
+	suffix := name[lastDash+1:]
+	if len(suffix) < 3 {
+		// Suffix is variable length but we cutoff at 3+ characters
+		return ""
+	}
+
+	if !stringinRuneset(suffix, Digits) && !stringinRuneset(suffix, KubeAllowedEncodeStringAlphaNums) {
+		// Invalid suffix
 		return ""
 	}
-	return match[1]
+
+	return name[:lastDash]
+}
+
+// isStringinRuneset tests whether all runes of a string are in a given subset
+// returns false if any rune in the string is not found in the subset
+func stringinRuneset(name, subset string) bool {
```

Comment:
IMHO this utility function does not belong to this file

Reply:
moved it to `utils` package, thanks!